data-loader/core/src/main/java/com/scalar/db/dataloader/core/tablemetadata/TableMetadataService.java

```diff
@@ -18,7 +19,17 @@
 public class TableMetadataService {
 
   private final DistributedStorageAdmin storageAdmin;
+  private final DistributedTransactionAdmin transactionAdmin;
```

Comment:
This class seems the union of `TableMetadataService` for storage and for transaction with some internal `if` branches. How about making `TableMetadataService` abstract and creating 2 child classes that return the TableMetadata for storage or transaction from something like `getTableMetadataInternal()`?

Reply:
@komamitsu san, 
I have split up table metadata service as suggested.
Thank you.